### ISSUE TITLE
Secrets files mounting broken after upgrade to 4.36.0 (WSL2)

### ISSUE THREAD

**Issue by das-peter** [docker/for-win#14434]
### Description

After updating to Docker Desktop 4.36.0 using secrets with files leads to following error:
```
docker compose up -d
[+] Running 0/0
 ⠋ Container test-alpine-1  Creating                                                                                                                                                                                                                              0.0s 
Error response from daemon: path /var/www/test/wooot is mounted on / but it is not a shared mount.
```

A downgrade (`\DockerDesktopInstaller.exe install --disable-version-check`) to Docker Desktop [4.35.1](https://docs.docker.com/desktop/release-notes/#4351)  "fixes" the issue.
A repeated upgrade didn't change anything.

### Reproduce

Following docker compose fails for me after updating to Docker Desktop 4.36.0:
1. Create a `docker-compose.yml` file with the following content:
```yml
secrets:
    foobar:
        file: ./wooot

services:
    alpine:
        image: alpine
        secrets:
            - foobar
```
2. Create a `wooot` file in the same directory as `docker-compose.yml` 
3. Run `docker compose up -d` in the directory where `docker-compose.yml`  is located

### Expected behavior

File based secrets should be mounted to `/var/run` as documented here https://docs.docker.com/compose/how-tos/use-secrets/#simple


### docker version

```bash
Client: Docker Engine - Community
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:41:03 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Desktop  ()
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Fri Sep 20 11:41:11 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.21
  GitCommit:        472731909fa34bd7bc9c087e4c27943f9835f111
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    27.3.1
 Context:    default
 Debug Mode: false
 Plugins:
  ai: Ask Gordon - Docker Agent (Docker Inc.)
    Version:  v0.1.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-ai
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.18.0-desktop.2
    Path:     /usr/local/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.30.3-desktop.1
    Path:     /usr/local/lib/docker/cli-plugins/docker-compose
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.37
    Path:     /usr/local/lib/docker/cli-plugins/docker-debug
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     /usr/local/lib/docker/cli-plugins/docker-desktop
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     /usr/local/lib/docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.27
    Path:     /usr/local/lib/docker/cli-plugins/docker-extension
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     /usr/local/lib/docker/cli-plugins/docker-feedback
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.4.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-sbom
  scout: Docker Scout (Docker Inc.)
    Version:  v1.15.0
    Path:     /usr/local/lib/docker/cli-plugins/docker-scout
WARNING: Plugin "/usr/local/lib/docker/cli-plugins/docker-scan" is not valid: failed to fetch metadata: fork/exec /usr/local/lib/docker/cli-plugins/docker-scan: no such file or directory

Server:
 Containers: 115
  Running: 25
  Paused: 0
  Stopped: 90
 Images: 65
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 nvidia
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 472731909fa34bd7bc9c087e4c27943f9835f111
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.167.4-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 5
 Total Memory: 23.48GiB
 Name: docker-desktop
 ID: 9b33a41f-afc7-48b2-a23f-b1dffffac70b
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=unix:///var/run/docker-cli.sock
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

553B5EAC-88C4-48F5-96A0-73CD956EC9F5/20241119192925

### Additional Info

_No response_

**Comment by naskoff** [comment]
+1

Exactly same situation 

Docker Desktop 4.36.0 (175267) is currently the newest version available.

`Error response from daemon: path ......./.secrets/user-password is mounted on / but it is not a shared mount.`

**Comment by scharris** [comment]
Same problem here, simple usage of secrets as described in docs. Was working prior to update to 4.36.0 but now gives error:

"Error response from daemon: path <my-file-redacted> is mounted on / but it is not a shared mount."

**Comment by sebastian-iancu** [comment]
+1
I encountered the same problem, the only solution was to downgrade to 4.35.1.
Hopefully, this will be fixed soon.

**Comment by larkiee** [comment]
i have exactly the same problem with secrets in docker compose

**Comment by radoslawkoziol** [comment]
We have the same problem

**Comment by emaborsa** [comment]
If you dont want to downgrade and want to wait the fixed release, you can mount it using `sudo mount --make-shared /`

**Comment by vipex** [comment]
Seems it has been fixed with version `4.37.0`

**Comment by das-peter** [comment]
> Seems it has been fixed with version 4.37.0

Thanks for letting us know. Just installed and can confirm.
I think it's fair to close this as complete.
